packages/shared/src/index.ts

```diff
@@ -1 +1,26 @@
+import { Query as RustQuery } from "./shared/shared";
 export * from "./shared/shared";
+
+/**
+ *  CallWithTimeout - calls a function with specified timeout
+ */
+const cwt =
+  <U extends unknown[], T>(fn: (...args: U) => Promise<T>, timeout = 5) =>
+  (...args: U): Promise<T> => {
+    return new Promise(async (resolve, reject) => {
+      const t = setTimeout(() => {
+        reject(`Balance query timed out after ${timeout}s.`);
```

Comment:
Maybe we could do something like [promiseWithTimeout](https://github.com/anoma/namada-interface/blob/48c0595e588c517c233dd4cb0615a08160aecbea/packages/utils/src/async/index.ts#L1-L22) where we pass in the error message as an arg to display, as these are not all balance queries?

Reply:
Ah yeah :) Focuesd to much on balances :+1: